command/debug.go

```diff
@@ -981,3 +991,29 @@ func (c *DebugCommand) captureError(target string, err error) {
 	})
 	c.errLock.Unlock()
 }
+
+func (c *DebugCommand) writeLogs(ctx context.Context) error {
+	out, err := os.Create(filepath.Join(c.flagOutput, "vault.log"))
+	if err != nil {
+		return err
+	}
+	logCh, err := c.cachedClient.Sys().Monitor(ctx, "trace")
+	if err != nil {
+		return err
+	}
+
+	for {
+		select {
+		case log, ok := <-logCh:
+			if !ok {
+				return nil
+			}
+			_, err = out.WriteString(log)
+			if err != nil {
+				return err
+			}
+		case <-ctx.Done():
+			return nil
+		}
+	}
```

Comment:
What causes this loop to terminate other than errors? I don't think the context will time out, will it?

Reply:
Yes, the context passed in has a timeout.